test/conformance/parallel/ptp.go

```diff
@@ -113,6 +116,78 @@ func testPtpSlaveClockSync(fullConfig testconfig.TestConfig, testParameters ptpt
 	// Actual implementation
 }
 
+// This test will run for configured minutes or until failure_threshold reached,
+// whatever comes first. A failure_threshold is reached each time the cpu usage
+// of any of the ptp pods (daemonset & operator) is higher than the expected one,
+// which is hardcoded to 15milliCores. The cpu usage is retrieved once per minute.
+func testPtpCpuUtilization(fullConfig testconfig.TestConfig, testParameters ptptestconfig.PtpTestConfig) {
+	const (
+		minimumFailureThreshold  = 1
+		cpuUsageCheckingInterval = 1 * time.Minute
+		milliCoresThreshold      = ptptestconfig.PtpDefaultMilliCoresUsageThreshold
+	)
+
+	logrus.Debugf("CPU Utilization TC Config: %+v", testParameters.SoakTestConfig.CpuUtilization)
+
+	params := testParameters.SoakTestConfig.CpuUtilization.TestSpec
+	if !params.Enable {
+		Skip("skip the test - the test is disabled")
+		return
+	}
+
+	// Create timer channel for test case timeout.
+	testCaseDuration := time.Duration(params.Duration) * time.Minute
+	tcEndChan := time.After(testCaseDuration)
+
+	// Create ticker for cpu usage checker function.
+	cpuUsageCheckTicker := time.NewTicker(cpuUsageCheckingInterval)
+
+	// Set failureThresold limit number.
+	failureThreshold := minimumFailureThreshold
+	if params.FailureThreshold > minimumFailureThreshold {
+		failureThreshold = params.FailureThreshold
+	}
+
+	logrus.Infof("Running TC for %s (failure threshold: %d)", testCaseDuration.String(), failureThreshold)
```

Comment:
Could we replace TC with test please? It may mean anything total compensation, terminal condition, transfer certificate etc etc. ;)

Reply:
Done.